crates/services/txpool/src/txpool.rs

```diff
@@ -233,37 +217,26 @@ where
         self.remove_by_tx_id(tx_id)
     }
 
-    fn verify_tx_min_gas_price(&mut self, tx: &Transaction) -> Result<(), Error> {
-        let price = match tx {
-            Transaction::Script(script) => script.price(),
-            Transaction::Create(create) => create.price(),
-            Transaction::Mint(_) => unreachable!(),
-        };
-        if self.config.metrics {
-            // Gas Price metrics are recorded here to avoid double matching for
-            // every single transaction, but also means metrics aren't collected on gas
-            // price if there is no minimum gas price
-            TXPOOL_METRICS.gas_price_histogram.observe(price as f64);
-        }
-        if price < self.config.min_gas_price {
-            return Err(Error::NotInsertedGasPriceTooLow)
-        }
-        Ok(())
-    }
-
     #[tracing::instrument(level = "info", skip_all)]
     /// Import a set of transactions from network gossip or GraphQL endpoints.
     pub fn insert(
         &mut self,
         tx_status_sender: &TxStatusChange,
-        txs: &[Arc<Transaction>],
+        txs: Vec<anyhow::Result<CheckedTransaction>>,
```

Comment:
I know that you try not to break the old behavior. But I think it is better to pass only transactions which passed the validation. Otherwise, we call `lock` on the `TxPool`, do nothing, and unlock it. If we have many `submit` requests with invalid transactions, each of them will block the `TxPool` while we can avoid it.

Reply:
good point!